canonical/ubuntu.com
# Infer preferred language from browser and send to marketo

## Done


- Adds logic to dynamic-forms.js to get the user preferred language from their browser and attach it to the marketo payload
- Makes the function available to all forms
- Incorporates the function [here](https://github.com/canonical/ubuntu.com/blob/main/templates/base_index.html#L1342) into it
- Updates the marketo_submit function in views.py. Adds the preferredLangauge field to the enrichment_payload
- Updates the [lead enrichment](https://engage-sj.marketo.com/?munchkinId=066-EOV-335#/mktform/4198/overview/details) form to accept a preferredLangauge input
- Adds the appropriate input fields to all forms

## QA

- Fill out a form and submit it
- Check the payload has the appropriate 'preferredLangauge' field added and it matches your browsers language setting
- Check the field has been updated on the marketo lead
- Check every form had the hidden input field `preferredLanguage`. To find every form I searched for 'button type="submit"' and manually added the input field to each result that was a form.

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-1436
